autoload/lsc/reference.vim

```diff
@@ -138,30 +138,40 @@ function! s:showHover(result) abort
     let contents = [contents]
   endif
   let lines = []
+  let filetype = 'markdown'
   for item in contents
     if type(item) == type({})
       let l:lines += split(item.value, "\n")
+      if has_key(item, 'language')
+        let l:filetype = item.language
+      elseif has_key(item, 'kind')
+        let l:filetype = item.kind
+      endif
     else
       let l:lines += split(item, "\n")
     endif
   endfor
   if get(g:, 'lsc_hover_popup', v:true) 
         \ && (exists('*popup_atcursor') || exists('*nvim_open_win'))
     call s:closeHoverPopup()
-    call s:openHoverPopup(l:lines)
+    call s:openHoverPopup(l:lines, l:filetype)
   else
     call lsc#util#displayAsPreview(lines, function('lsc#util#noop'))
   endif
 endfunction
 
-function! s:openHoverPopup(lines) abort
+function! s:openHoverPopup(lines, filetype) abort
   " Sanity check, if there is no hover text then don't waste resources creating an
   " empty popup.
   if len(a:lines) == 0
     return
   endif
   if has('nvim')
     let buf = nvim_create_buf(v:false, v:true)
+    call nvim_buf_set_option(buf, 'synmaxcol', 0)
+    if exists('g:lsc_enable_popup_syntax') && g:lsc_enable_popup_syntax
```

Comment:
I'm inclined to skip the configuration option and always enable this. Can you think of any reason someone _wouldn't_ want this?

Reply:
Without this, it causes visible syntax problems for extremely long syntax lines, like exist in Flutter API. That is if a user sets synmaxcol, like I do.

I should provide an image of this to show why I did this.
I am on my iPad at the moment, so I will show you later today the issue this line resolves.

Ideally the nvm floating window should clear all settings and start blank, it mostly does but not quite completely.

Reply:
Oops, I misunderstood your comment here, I thought you were talking about the `synmaxcol` change.

You were actually talking about the option itself.

I believe most people would prefer the syntax colouring. 

However, a couple issues:

- The Dart hover segments had blank lines with spaces, which with [plasticboy's Markdown plugin](https://github.com/plasticboy/vim-markdown) were being highlighted with the `mdkLineBreak` syntax group. With my own [moonfly](https://github.com/bluz71/vim-moonfly-colors) theme those were being highlighted with normal colouring (rather than popup colors) hence, the hover window looked like it had swiss cheese holes. I needed to modify my Vim theme to change `mdkLineBreak` color. Will other themes have the same issue? I suspect yes with Dart LSP hovering.

- We default to the markdown syntax, which should be ok most of the time even with plain-text hover details. Noting that Dart LSP returns a plain-string with no kind hint, but the details really are markdown. But it is conceivable that a language server may return real unhinted plaintext, what if the plain text uses a single backtick? The formatting will end up broken.

I would definitely leave the option available just as a failsafe.

The question is default to 'on' or default to 'off'?
I would lean to default 'on', though this PR defaulted it to 'off'.

Have you played with this PR? Did you get the Swiss cheese effect with Dart/Flutter hovering?

Actually let me attach an image showing the issue:

<img width="567" alt="Dart_LSP_Hover" src="https://user-images.githubusercontent.com/11382509/70198937-6c568780-1764-11ea-9c63-e3941ab8c362.png">

Notice the little black bits along the left-hand edge? I fixed it in my theme. Ideally the Dart hover details should not ship empty lines with a couple leading spaces.

Reply:
I played around with this today and I really like it. Let's ship it with the option defaulted to enabled.

```suggestion
    if get(g:, 'lsc_enable_popup_syntax', v:true)
```